### PR TITLE
fix: Check for NULL in binary image cache

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Properly validate the binary image name when converting it to an NSString to avoid crashes.

## :bulb: Motivation and Context

I came across this when looking at https://github.com/getsentry/sentry-cocoa/pull/3468.

## :green_heart: How did you test it?
Unit tests.

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: Sources/Sentry/SentryBinaryImageCache.m [hunk 1]
```diff
@@ -35,8 +36,21 @@ - (void)stop
 
 - (void)binaryImageAdded:(const SentryCrashBinaryImage *)image
 {
+    if (image->name == NULL) {
```
Comment (armcknight) [inline Sources/Sentry/SentryBinaryImageCache.m:39]: Is it possible for `image` to also be `NULL`? If `nextNode->image` for whatever reason in NULL, then we could pass a NULL pointer and dereferencing here would crash. https://github.com/getsentry/sentry-cocoa/blob/24c239b8e33dc88bfc70f588069a029eb00b4165/Sources/SentryCrash/Recording/SentryCrashBinaryImageCache.c#L143-L147

File: Sources/Sentry/SentryBinaryImageCache.m [hunk 2]
```diff
@@ -35,8 +36,21 @@ - (void)stop
...
+        SENTRY_LOG_WARN(@"The image name was NULL. Can't add image to cache.");
+        return;
+    }
+
+    NSString *imageName = [NSString stringWithCString:image->name encoding:NSUTF8StringEncoding];
```
Comment (armcknight) [inline Sources/Sentry/SentryBinaryImageCache.m:44]: Are we sure `image->name` will never be NULL? That can crash this constructor.

**Review by armcknight — COMMENTED** [review]
Couple questions on possible additional NULL checks we might need.

File: Sources/Sentry/SentryBinaryImageCache.m (hunk 1 above)
Comment (philipphofmann) [reply Sources/Sentry/SentryBinaryImageCache.m:43]: I think it shouldn't but a check for NULL does not harm. Thanks for pointing that out.

Comment (philipphofmann) [reply Sources/Sentry/SentryBinaryImageCache.m:43]: I think it shouldn't but a check for NULL does not harm. Thanks for pointing that out.

File: Sources/Sentry/SentryBinaryImageCache.m (hunk 2 above)
Comment (philipphofmann) [reply Sources/Sentry/SentryBinaryImageCache.m:48]: I added a check for that above `if (image->name == NULL)`. If it's `NULL` we return.

**Review by brustolin — APPROVED** [review]
LGTM

But, can this ever happen, the system calling binaryWasAdded but passing NULL?

**Pull request merged**